core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -287,12 +288,29 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
   def recordAndGetThrottleTimeMs(session: Session, clientId: String, value: Double, timeMs: Long): Int = {
     var throttleTimeMs = 0
     val clientSensors = getOrCreateQuotaSensors(session, clientId)
+    val clientMetric = metrics.metrics().get(clientRateMetricName(clientSensors.metricTags))
+
+    // A helper function to deal with Some(null) in Option chain
+    def nullResultAsNone[X, Y](f: X => Y): X => Option[Y] = (x: X) => Option(f(x))
+    // Quotas may 1) not be configured or 2) not be applicable to all metrics,
+    // so it's necessary to check for the bound's validity beforehand
+    def quotaBound =
```

Comment:
Since the clientMetric.config may have a null quota field, but the quota's bound field should always be available, can we simply check if the clientclientMetric.config.quota field is null or not?
We should always report even if its value is 0. Also, we shouldn't worry about the isNaN case.

Reply:
I think you're right on 0.  Fixed it.
But I think NaN is a safety check here so I still kept it.